src/transformers/quantizers/quantizer_hqq.py

```diff
@@ -122,21 +139,47 @@ def create_quantized_param(
             from hqq.core.quantize import HQQLinear
 
         module, tensor_name = get_module_from_name(model, param_name)
-
-        layer_name = param_name.replace(".weight", "").replace(".bias", "")
+        layer_name = ".".join(param_name.split(".")[:-1])
         parent_module = find_parent(model, layer_name)
         node = layer_name.split(".")[-1]
 
-        # Step 0: set module state_dict
-        module_state_dict = {key.split(".")[-1]: state_dict[key] for key in state_dict if layer_name in key}
+        # set module state_dict
+        module_state_dict = {}
+        for k, v in state_dict.items():
+            if layer_name + "." in k:
+                module_state_dict[k.split(".")[-1]] = v
+                if unexpected_keys is not None and k in unexpected_keys:
+                    unexpected_keys.remove(k)
+
+        if self.pre_quantized:
+            if isinstance(module, (torch.nn.Linear, HQQLinear)):
+                hqq_layer = HQQLinear(
+                    linear_layer=None,
+                    quant_config=None,
+                    compute_dtype=self.torch_dtype,
+                    device=target_device,
+                )
+
+            hqq_layer.axis = None
+            hqq_layer.channel_wise = None
+            hqq_layer.load_state_dict(module_state_dict)
```

Comment:
In my PR, I have two cases: 
- if module is HQQLinear, we do a return 
- if module is a Linear, we replace it by hqqlinear + load_state dict 

Here, we do the same. This will make the loading way slower since we load the layer multiple times and it is probably what you are experiencing.

Reply:
Reverting back to that version :+1:

Reply:
Maybe sure to push the commit ! To me, it's still not reverted.

Reply:
it;s here
https://github.com/huggingface/transformers/blob/cbe219fe3ab5694f1fd2ace6597cb74bd8324d9a/src/transformers/quantizers/quantizer_hqq.py#L154-L178

Reply:
Apologies for the confusion, it actually works with the reverted changes: https://github.com/huggingface/transformers/pull/33141/commits/2bb974c5e4faae8e110fa3bb42f22b0a1e5f7736